src/include/function/function_collection.h

```diff
@@ -1,16 +1,21 @@
 #pragma once
 
+#include "catalog/catalog_entry/catalog_entry_type.h"
 #include "function.h"
 
+using namespace kuzu::catalog;
+
 namespace kuzu {
 namespace function {
 
 using get_function_set_fun = std::function<function_set()>;
 
 struct FunctionCollection {
+    get_function_set_fun getFunctionSetFunc;
+
     const char* name;
 
-    get_function_set_fun getFunctionSetFunc;
+    CatalogEntryType catalogEntryType;
```

Comment:
Why do we need this field? i don't see the place it has been assigned.

Reply:
It is assigned here: https://github.com/kuzudb/kuzu/pull/3134/files#diff-f0f3d8e926fc515de278119a26578e49adbb69f08b7cb4d24fd546507b4981c8L21

Reply:
I think you can still use ScalarCatalogEntry instead because you are using nullptr to check the end.